crates/biome_js_analyze/src/lint/nursery/no_document_cookie.rs

```diff
@@ -0,0 +1,111 @@
+use biome_analyze::{context::RuleContext, declare_lint_rule, Rule, RuleDiagnostic, RuleSource};
+use biome_console::markup;
+use biome_js_syntax::{
+    global_identifier, AnyJsAssignment, AnyJsAssignmentPattern, JsAssignmentExpression,
+};
+use biome_rowan::AstNode;
+
+use crate::services::semantic::Semantic;
+
+declare_lint_rule! {
+    /// Disallow use `document.cookie` directly.
+    ///
+    /// It's not recommended to use document.cookie directly as it's easy to get the string wrong.
+    /// Instead, you should use the [Cookie Store API](https://developer.mozilla.org/en-US/docs/Web/API/CookieStore) or a [cookie library](https://www.npmjs.com/search?q=cookie).
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```js,expect_diagnostic
+    /// document.cookie = "foo=bar";
+    /// ```
+    ///
+    /// ```js,expect_diagnostic
+    /// document.cookie += "; foo=bar";
+    /// ```
+    ///
+    /// ### Valid
+    ///
+    /// ```js
+    /// await cookieStore
+    ///   .set({
+    ///     name: "foo",
+    ///     value: "bar",
+    ///     expires: Date.now() + 24 * 60 * 60,
+    ///     domain: "example.com",
+    /// })
+    /// ```
+    ///
+    /// ```js
+    /// const array = document.cookie.split("; ");
+    /// ```
```

Comment:
You should move this to the top, because it's the most relevant example that shows that **assignments** aren't valid, but it's still possible to call functions over for `document.cookie`

Reply:
Fixed at https://github.com/biomejs/biome/pull/4204/commits/b46d4dcb2f10206fe801534b9fc39cf92be8a4dc